packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/textBoxMapping/MappingTargetCellProvider.ts

```diff
@@ -0,0 +1,57 @@
+import { TextLayoutCellBase } from '../textLayout/types';
+import { TextNormalizer } from '../common/TextNormalizer';
+import { CellProvider } from './CellProvider';
+import { END } from '../common/textSpanUtils';
+
+export class MappingTargetBoxProvider {
+  private readonly cellProvider: CellProvider;
+  private current: {
+    nextCellIndex: number;
+    normalizer: TextNormalizer;
+    leadingSpaces: number;
+  } | null = null;
+
+  constructor(cells: TextLayoutCellBase[]) {
+    this.cellProvider = new CellProvider(cells);
+  }
+
+  hasNext() {
```

Comment:
I wonder: does it make sense to convert this and `MappingSourceTextProvider` to a [JS iterable](https://dev.to/xenoxdev/javascript-jungle-convert-any-object-to-iterable-40l6)?

Reply:
Let me try :)

Reply:
I tried, but I found that there is similarities between the class and JS iterable. But the class needs explicit user action `skip` or `consume` to move the _current index_ to the next. So, I'd like to keep this as-is.

We'll need `"downlevelIteration": true` in tsconfig.json to use user-defined iterator 😄